File: src/main/java/com/bajdcc/LALR1/ui/UIRemotePanel.java
package com.bajdcc.LALR1.ui;

import com.bajdcc.LALR1.ui.drawing.UIRemoteGraphics;

import javax.swing.*;
import java.awt.*;

/**
 * 【界面】远程渲染界面
 *
 * @author bajdcc
 */
public class UIRemotePanel extends JPanel {

	private UIRemoteGraphics graphics;

	public static int w = 890;
	public static int h = 655;

	public UIRemotePanel() {
		this.graphics = new UIRemoteGraphics(w, h);
		this.setFocusable(false);
	}

	public UIRemoteGraphics getUIGraphics() {
		return graphics;
	}

	public void paint(Graphics g) {
		graphics.paint((Graphics2D) g);
	}
}

File: src/main/kotlin/com/bajdcc/LALR1/ui/UIRemotePanel.kt
package com.bajdcc.LALR1.ui

import com.bajdcc.LALR1.ui.drawing.UIRemoteGraphics
import java.awt.Graphics
import java.awt.Graphics2D
import javax.swing.JPanel

/**
 * 【界面】远程渲染界面
 *
 * @author bajdcc
 */
class UIRemotePanel : JPanel() {

    val uiGraphics: UIRemoteGraphics

    init {
        this.uiGraphics = UIRemoteGraphics(w, h)
        this.isFocusable = false
    }

    override fun paint(g: Graphics) {
        uiGraphics.paint(g as Graphics2D)
    }

    companion object {

        var w = 890
        var h = 655
    }
}
